src/main/webapp/app/shared/layouts/profiles/profile.service.ts

```diff
@@ -26,17 +26,21 @@ export class ProfileService {
                         const data = res.body!;
                         const profileInfo = new ProfileInfo();
                         profileInfo.activeProfiles = data.activeProfiles;
-                        const displayRibbonOnProfiles = data['display-ribbon-on-profiles'].split(',');
 
                         this.mapGuidedTourConfig(data, profileInfo);
                         this.mapAllowedOrionVersions(data, profileInfo);
                         this.mapTestServer(data, profileInfo);
                         ProfileService.mapSaml2Config(data, profileInfo);
 
                         if (profileInfo.activeProfiles) {
-                            const ribbonProfiles = displayRibbonOnProfiles.filter((profile: string) => profileInfo.activeProfiles.includes(profile));
-                            if (ribbonProfiles.length !== 0) {
-                                profileInfo.ribbonEnv = ribbonProfiles[0];
+                            if (data['display-ribbon-on-profiles']) {
+                                const displayRibbonOnProfiles = data['display-ribbon-on-profiles'].split(',');
+                                const ribbonProfiles = displayRibbonOnProfiles.filter((profile: string) => profileInfo.activeProfiles.includes(profile));
+                                if (ribbonProfiles.length !== 0) {
+                                    profileInfo.ribbonEnv = ribbonProfiles[0];
+                                }
```

Comment:
Shouldn't you set it to '' in an else statement here as well? I don't know the code but it looks like that from the previous version.

Reply:
Yes, I think you are correct. I replaced the else with an undefined check of `ribbonEnv`, if it is indeed undefined I set it to `''`. Thank you for spotting this 👍